charmcraft/providers/_get_provider.py

```diff
@@ -17,12 +17,55 @@
 """Build environment provider support for charmcraft."""
 
 import logging
+import os
+import sys
+
+from charmcraft.snap import get_snap_configuration
+from charmcraft.cmdbase import CommandError
+from charmcraft.env import is_charmcraft_running_from_snap, is_charmcraft_running_in_developer_mode
 
 from ._lxd import LXDProvider
+from ._multipass import MultipassProvider
 
 logger = logging.getLogger(__name__)
 
 
+def _get_platform_default_provider() -> str:
+    if sys.platform == "linux":
+        return "lxd"
+
+    return "multipass"
+
+
 def get_provider():
-    """Get the configured or appropriate provider for the host OS."""
-    return LXDProvider()
+    """Get the configured or appropriate provider for the host OS.
+
+    If platform is not Linux, use Multipass.
+
+    If platform is Linux:
+    (1) use provider specified with CHARMCRAFT_PROVIDER if running
+        in developer mode,
+    (2) use provider specified with snap configuration if running
+        as snap,
+    (3) default to platform default (LXD on Linux).
+
+    :return: Provider instance.
+    """
+    provider = None
+
+    if is_charmcraft_running_in_developer_mode():
```

Comment:
personally I would restrict this to snapctl for now, is there a need for this envirnonment variable now? What are the user stories solved with it?

Reply:
It's the developer story, which is why I restrict the env var to developer mode (CHARMCRAFT_DEVELOPER=y).  When developing on it, it prevents me from having to rebuild every change as a snap...